### PR TITLE
Bump aws-sdk-go to v1.32.7

### LINKED ISSUE #3510: Teleport fails to use EKS ServiceAccount IAM role
### Description
I deployed Teleport into an Amazon EKS cluster in high-availability mode (using DynamoDB and S3) based on your [example Terraform](https://github.com/gravitational/teleport/tree/v4.2.8/examples/aws/terraform) and [chart](https://github.com/gravitational/teleport/tree/v4.2.8/examples/chart/teleport), modified to work in EKS. I assigned the Teleport auth IAM role to the Teleport Kubernetes ServiceAccount.

**What happened**:

Teleport auth service failed to start because it could not access the DynamoDB table

**What you expected to happen**:

Teleport pod would provide auth and proxy services

**How to reproduce it (as minimally and precisely as possible)**:

### Environment
- Teleport version (use `teleport version`): 4.2.8
- Tsh version (use `tsh version`): NA
- OS (e.g. from `/etc/os-release`): NA

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS EKS

**Relevant Debug Logs If Applicable**

```
Original Error: *awserr.requestError AccessDeniedException: User: arn:aws:sts::<redacted>:assumed-role/example-eks-us-east-2c-workers/i-0f85ef29911b013a3 is not authorized to perform: dynamodb:DescribeTable on resource: arn:aws:dynamodb:us-east-2:<redacted>:table/example-teleport
        status code: 400, request id: <redacted>
Stack Trace:
        /gopath/src/github.com/gravitational/teleport/lib/backend/dynamo/dynamodbbk.go:575 github.com/gravitational/teleport/lib/backend/dynamo.(*DynamoDBBackend).getTableStatus
        /gopath/src/github.com/gravitational/teleport/lib/backend/dynamo/dynamodbbk.go:236 github.com/gravitational/teleport/lib/backend/dynamo.New
        /gopath/src/github.com/gravitational/teleport/lib/service/service.go:2331 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthStorage
        /gopath/src/github.com/gravitational/teleport/lib/service/service.go:906 github.com/gravitational/teleport/lib/service.(*TeleportProcess).initAuthService
        /gopath/src/github.com/gravitational/teleport/lib/service/service.go:648 github.com/gravitational/teleport/lib/service.NewTeleport
        /gopath/src/github.com/gravitational/teleport/e/lib/pro/process.go:41 github.com/gravitational/teleport/e/lib/pro.NewTeleport
        /gopath/src/github.com/gravitational/teleport/e/tool/teleport/main.go:35 main.run.func1
        /gopath/src/github.com/gravitational/teleport/lib/service/service.go:401 github.com/gravitational/teleport/lib/service.Run
        /gopath/src/github.com/gravitational/teleport/e/tool/teleport/main.go:43 main.run
        /gopath/src/github.com/gravitational/teleport/e/tool/teleport/main.go:27 main.main
        /opt/go/src/runtime/proc.go:212 runtime.main
        /opt/go/src/runtime/asm_amd64.s:1358 runtime.goexit
User Message: initialization failed

```

### PR DESCRIPTION
**Pull request by awly**
This should fix https://github.com/gravitational/teleport/issues/3510 and likely pick up other improvements.

### CODE REVIEW

File: Gopkg.toml
Comment (benarent) [inline Gopkg.toml:72]: ^ Version number needs to be bumped.

Comment (awly) [reply Gopkg.toml]: I set it to the version from https://github.com/gravitational/teleport/issues/3510#issuecomment-607774810
Note that it's the _minimum_ version, the exact version is in `Gopkg.lock`.
Anyway, no reason not to make it higher so updated to match `Gopkg.lock`.

**Review by webvictim — APPROVED** [review]
LGTM. We should backport this to the 4.3 branch and cut `4.3.0-alpha.2` with this included so we can use it for the test plan.

File: Gopkg.toml
Comment (benarent) [reply Gopkg.toml]: Thanks for the clarification, I just scanned the PR earlier. Makes sense.

**Comment by benarent** [comment]
@awly Please ping me when this is done, as I've EKS to complete for the test plan.

**Review by fspmarshall — APPROVED** [review]
LGMT

*Disclaimer:* AWS is definitely not my area of expertise.

**Comment by awly** [comment]
@russjones need a codeowners approval here

**Review by russjones — APPROVED** [review]

**Pull request merged**
